stubs/openpyxl/openpyxl/worksheet/worksheet.pyi

```diff
@@ -24,106 +31,106 @@ class Worksheet(_WorkbookChild):
     PAPERSIZE_A5: str
     ORIENTATION_PORTRAIT: str
     ORIENTATION_LANDSCAPE: str
-    def __init__(self, parent, title: Incomplete | None = ...) -> None: ...
+    def __init__(self, parent: Workbook, title: str | None = ...) -> None: ...
     @property
-    def sheet_view(self): ...
+    def sheet_view(self) -> SheetView: ...
     @property
-    def selected_cell(self): ...
+    def selected_cell(self) -> Cell: ...
     @property
-    def active_cell(self): ...
+    def active_cell(self) -> Cell: ...
     @property
-    def page_breaks(self): ...
+    def page_breaks(self) -> PageBreak: ...
     @property
-    def show_gridlines(self): ...
+    def show_gridlines(self) -> Any: ...
     @property
-    def show_summary_below(self): ...
+    def show_summary_below(self) -> Any: ...
     @property
-    def show_summary_right(self): ...
+    def show_summary_right(self) -> Any: ...
     @property
-    def freeze_panes(self): ...
+    def freeze_panes(self) -> Any: ...
     @freeze_panes.setter
     def freeze_panes(self, topLeftCell: Incomplete | None = ...) -> None: ...
-    def cell(self, row, column, value: Incomplete | None = ...): ...
-    def __getitem__(self, key): ...
-    def __setitem__(self, key, value) -> None: ...
-    def __iter__(self): ...
-    def __delitem__(self, key) -> None: ...
+    def cell(self, row: int, column: int, value: str | None = ...) -> Cell: ...
+    def __getitem__(self, key: str) -> Cell | tuple[Cell]: ...
+    def __setitem__(self, key: str, value: str) -> None: ...
+    def __iter__(self) -> Iterator[Cell]: ...
+    def __delitem__(self, key: str) -> None: ...
     @property
-    def min_row(self): ...
+    def min_row(self) -> int: ...
     @property
-    def max_row(self): ...
+    def max_row(self) -> int: ...
     @property
-    def min_column(self): ...
+    def min_column(self) -> int: ...
     @property
-    def max_column(self): ...
-    def calculate_dimension(self): ...
+    def max_column(self) -> int: ...
+    def calculate_dimension(self) -> str: ...
     @property
-    def dimensions(self): ...
+    def dimensions(self) -> str: ...
     def iter_rows(
         self,
-        min_row: Incomplete | None = ...,
-        max_row: Incomplete | None = ...,
-        min_col: Incomplete | None = ...,
-        max_col: Incomplete | None = ...,
+        min_row: int | None = ...,
+        max_row: int | None = ...,
+        min_col: int | None = ...,
+        max_col: int | None = ...,
         values_only: bool = ...,
-    ): ...
+    ) -> Generator[Iterable[Cell], None, None]: ...
```

Comment:
The first parameter of `Generator` is the type that the generator yields. So this should be `Generator[Cell, None, None]` rather than `Generator[Iterable[Cell], None, None]`.

Additionally, this method will yield the _values_ of the cells, rather than the cells themselves, if `values_only=True` is passed. You could represent this using overloads.